### PR TITLE
feat: reorder columns from table options

### LINKED ISSUE #914: I can re-order columns using the Option Button
Related to https://github.com/twentyhq/twenty/issues/813

### PR DESCRIPTION
**Pull request by AdityaPimpalkar**
closes: #914

Two things to be fixed

1) MenuItem is turning transparent on dragging (not sure what's causing this)
2) The drag select on table is being trigged while dragging the options (maybe we can temporarily disable drag select when the options dropdown is opened) 

https://github.com/twentyhq/twenty/assets/13139771/3b9fef3d-0fba-4500-bb19-80850a2294c2

### CODE REVIEW

**Comment by lucasbordeau** [comment]
Nice work, could you please create a new recoil atom to activate/deactivate drag select ?

File: front/src/modules/ui/table/components/EntityTable.tsx [hunk 1]
```diff
@@ -88,6 +90,7 @@ type OwnProps = {
 
 export const EntityTable = ({ updateEntityMutation }: OwnProps) => {
   const tableBodyRef = useRef<HTMLDivElement>(null);
+  const [dragSelect, setDragSelect] = useRecoilState(dragSelectState);
```
Comment (lucasbordeau) [inline front/src/modules/ui/table/components/EntityTable.tsx:93]: Could you find a more self-explanatory naming, it should tell us what's happening. 

Maybe `isDragSelecting` or `isDraggingAndSelecting` ?

**Review by lucasbordeau — CHANGES_REQUESTED** [review]

File: front/src/modules/ui/table/components/EntityTable.tsx (hunk 1 above)
Comment (AdityaPimpalkar) [reply front/src/modules/ui/table/components/EntityTable.tsx:93]: Thanks @lucasbordeau! I always struggle with naming variables :')

**Comment by AdityaPimpalkar** [comment]
Please ignore the `packages/` changes on this PR, I think I messed up the merging. Apologies for that

**Review by lucasbordeau — APPROVED** [review]

**Comment by lucasbordeau** [comment]
Just the conflict to solve and we're good !

**Comment by charlesBochet** [comment]
Merging this one too, two big pieces @AdityaPimpalkar!

**Comment by charlesBochet** [comment]
@AdityaPimpalkar I've created a ticket here to make what you've done available and re-usable in the codebase. That would be great for future implementation. Would love if you could work on it! https://github.com/twentyhq/twenty/issues/1668

**Pull request merged**